cmd/batch-handlers.go

```diff
@@ -1671,11 +1676,13 @@ func (a adminAPIHandlers) CancelBatchJob(w http.ResponseWriter, r *http.Request)
 	}
 
 	j := BatchJobRequest{
-		ID:       jobID,
-		Location: pathJoin(batchJobPrefix, jobID),
+		ID: jobID,
 	}
 
-	j.delete(ctx, objectAPI)
+	if err := j.delete(ctx, objectAPI); err != nil {
+		writeErrorResponseJSON(ctx, w, toAPIError(ctx, err), r.URL)
+		return
+	}
 
```

Comment:
The idea here for not returning errors is to avoid clients retrying the operation again and again. It is not necessary TBH and we must simply garbage collect these. 

We should not waste time returning errors to the client because that is not a useful use of a client's time such as `mc`.  Users can move on to new things because it is a cancel operation. The actual job is already canceled, we are just deleting a job ID and its state() file. 

We should be more focussed on a more idempotent behavior.